go/geth_adapter/adapter_test.go

```diff
@@ -259,6 +259,166 @@ func TestRunContextAdapter_SnapshotHandling(t *testing.T) {
+				return adapter.GetTransientStorage(tosca.Address{}, tosca.Key{})
+			},
+		},
+		"balance": {
+			primingMock: func(stateDb *MockStateDb) {
+				stateDb.EXPECT().GetBalance(gomock.Any()).Return(uint256.NewInt(100))
+			},
+			want: tosca.NewValue(100),
+			functionCall: func(adapter *runContextAdapter) any {
+				return adapter.GetBalance(tosca.Address{})
+			},
+		},
+		"codeSize": {
+			primingMock: func(stateDb *MockStateDb) {
+				stateDb.EXPECT().GetCodeSize(gomock.Any()).Return(42)
+			},
+			want: 42,
+			functionCall: func(adapter *runContextAdapter) any {
+				return adapter.GetCodeSize(tosca.Address{})
+			},
+		},
+		"codeHash": {
+			primingMock: func(stateDb *MockStateDb) {
+				stateDb.EXPECT().GetCodeHash(gomock.Any()).Return(common.Hash{4})
+			},
+			want: tosca.Hash{4},
+			functionCall: func(adapter *runContextAdapter) any {
+				return adapter.GetCodeHash(tosca.Address{})
+			},
+		},
+		"code": {
+			primingMock: func(stateDb *MockStateDb) {
+				stateDb.EXPECT().GetCode(gomock.Any()).Return(tosca.Code{5, 6, 7})
+			},
+			want: tosca.Code{5, 6, 7},
+			functionCall: func(adapter *runContextAdapter) any {
+				return adapter.GetCode(tosca.Address{})
+			},
+		},
+		"committedStorage": {
+			primingMock: func(stateDb *MockStateDb) {
+				stateDb.EXPECT().GetCommittedState(gomock.Any(), gomock.Any()).Return(common.Hash{42})
+			},
+			want: tosca.Word{42},
+			functionCall: func(adapter *runContextAdapter) any {
+				return adapter.GetCommittedStorage(tosca.Address{}, tosca.Key{})
+			},
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			stateDb := NewMockStateDb(ctrl)
+			test.primingMock(stateDb)
+
+			adapter := &runContextAdapter{evm: &geth.EVM{StateDB: stateDb}}
+			got := test.functionCall(adapter)
+
+			if !reflect.DeepEqual(got, test.want) {
+				t.Errorf("Got wrong value %v, expected %v", got, test.want)
+			}
+		})
+	}
+}
+
+func TestRunContextAdapter_SettersForwardTheCorrectStateDbValues(t *testing.T) {
+	tests := map[string]struct {
+		primingMock  func(stateDb *MockStateDb)
+		functionCall func(adapter *runContextAdapter)
+	}{
+		"nonce": {
+			primingMock: func(stateDb *MockStateDb) {
+				stateDb.EXPECT().SetNonce(common.Address{0x42}, uint64(42), tracing.NonceChangeUnspecified)
+
+			},
+			functionCall: func(adapter *runContextAdapter) {
+				adapter.SetNonce(tosca.Address{0x42}, 42)
+			},
+		},
+		"storage": {
+			primingMock: func(stateDb *MockStateDb) {
+				stateDb.EXPECT().GetState(common.Address{0x42}, common.Hash{0x10}).Return(common.Hash{0x01})
+				stateDb.EXPECT().GetCommittedState(common.Address{0x42}, common.Hash{0x10}).Return(common.Hash{0x02})
+				stateDb.EXPECT().SetState(common.Address{0x42}, common.Hash{0x10}, common.Hash{0x03})
+			},
+			functionCall: func(adapter *runContextAdapter) {
+				adapter.SetStorage(tosca.Address{0x42}, tosca.Key{0x10}, tosca.Word{0x03})
```

Comment:
Idea for a future test: [SetStorage](https://github.com/0xsoniclabs/tosca/blob/de6bdd9fe6db79bd08f9bc739e364a1a2c250a25/go/geth_adapter/adapter.go#L413) returns a `tosca.StorageStatus`. It would be good to add tests checking this return value -- either through a few examples or exhaustive.

Reply:
Just noticed that this is covered in https://github.com/0xsoniclabs/tosca/pull/106 :+1: